src/PrestaShopBundle/Controller/Admin/Sell/Catalog/FeatureController.php

```diff
@@ -26,163 +26,155 @@
+    public function indexAction(Request $request)
     {
-        try {
-            $editableFeature = $this->getQueryBus()->handle(new GetFeatureForEditing((int) $featureId));
-        } catch (Exception $e) {
-            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
-
-            // @todo change route to features index when it's migrated
-            return $this->redirectToRoute('admin_features_create');
-        }
+        $this->setListFields();
+        $this->buildActionList();
+
+        $helperListConfiguration = $this->get('prestashop.core.bridge.helper_list_configuration_factory')->create([
+            'table' => self::TABLE,
+            'listId' => self::LIST_ID,
+            'className' => self::CLASS_NAME,
+            'controllerNameLegacy' => self::CONTROLLER_NAME_LEGACY,
+            'identifier' => self::IDENTIFIER,
+            'isJoinLanguageTableAuto' => true,
+            'defaultOrderBy' => 'position',
+            'fieldsList' => $this->controllerConfiguration->fieldsList,
+        ]);
 
-        if (!$this->isFeatureEnabled()) {
-            return $this->renderEditForm([
-                'showDisabledFeatureWarning' => true,
-                'editableFeature' => $editableFeature,
-            ]);
+        if ($request->request->has('submitResetfeature')) {
+            $this->get('prestashop.core.bridge.processor.reset_filter')->resetFilters($helperListConfiguration, $request);
         }
 
-        $featureFormBuilder = $this->get('prestashop.core.form.identifiable_object.builder.feature_form_builder');
-        $featureFormHandler = $this->get('prestashop.core.form.identifiable_object.handler.feature_form_handler');
+        $this->get('prestashop.core.bridge.processor.process_filter')->processFilter(
+            $request,
+            $helperListConfiguration
+        );
 
-        $featureForm = $featureFormBuilder->getFormFor($featureId);
-        $featureForm->handleRequest($request);
+        $this->get('prestashop.core.bridge.helper_list_bridge')->getList(
+            $helperListConfiguration,
+            $this->getContext()->language->id
+        );
 
-        try {
-            $handlerResult = $featureFormHandler->handleFor($featureId, $featureForm);
-
-            if ($handlerResult->isSubmitted() && $handlerResult->isValid()) {
-                $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
-
-                return $this->redirectToRoute('admin_features_edit', [
-                    'featureId' => $featureId,
-                ]);
-            }
-        } catch (Exception $e) {
-            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
-        }
-
-        return $this->renderEditForm([
-            'featureForm' => $featureForm->createView(),
-            'editableFeature' => $editableFeature,
-        ]);
+        return $this->renderSmarty(
```

Comment:
Perhaps we should make this work more similarly to `$this->render()`? (eg. be able to can pass along more variables, etc)

Reply:
It might be convenient indeed, but I guess we can always add optional parameters later if needed